core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceAdminService.scala

```diff
@@ -107,6 +110,60 @@ class WorkspaceAdminService(
       settings
     )
 
+  /**
+   * Admin endpoint to delete a workspace of type MC
+   */
+  def adminDeleteMcWorkspace(workspaceName: WorkspaceName): Future[Unit] =
+    asFCAdmin {
+      for {
+        // Get workspace to verify it's an MC workspace and to get its ID
+        workspaceOpt <- workspaceRepository.getWorkspace(workspaceName)
+        workspace = workspaceOpt.getOrElse(throw NoSuchWorkspaceException(workspaceName))
+        _ = if (workspace.workspaceType != WorkspaceType.McWorkspace) {
+          throw new RawlsExceptionWithErrorReport(
+            ErrorReport(StatusCodes.BadRequest, s"Workspace ${workspaceName} is not an MC workspace")
+          )
+        }
+
+        // Add the current caller to the workspace owner policy to ensure they have sufficient permissions
+        _ <- samDAO.admin.addUserToPolicy(
+          SamResourceTypeNames.workspace,
+          workspace.workspaceId,
+          SamWorkspacePolicyNames.owner,
+          ctx.userInfo.userEmail.value,
+          ctx
+        )
+
+        _ <- recursivelyDeleteSamResource(SamResourceTypeNames.workspace, workspace.workspaceId, ctx)
+        _ <- workspaceRepository.deleteWorkspace(workspaceName)
```

Comment:
this will fail with FK errors if there are any entities or submissions in the workspace … which should never happen for Azure workspaces. However, there was a window of time in which it was possible to create entities in an Azure workspace (they never appeared in the UI, but they could have been created). TL;DR we may find corner cases where this fails.